geocode.go

```diff
@@ -109,25 +109,28 @@ type Response struct {
 type Result struct {
 	Address      string         `json:"formatted_address"`
 	AddressParts []*AddressPart `json:"address_components"`
-	Geometry     *Geometry
-	Types        []string
+	Geometry     *Geometry      `json:"geometry"`
+	Types        []string       `json:"types"`
+	PartialMatch bool           `json:"partial_match,omitempty"`
+	PlaceId      string         `json:"place_id,omitempty"`
 }
 
 type AddressPart struct {
-	Name      string `json:"long_name"`
-	ShortName string `json:"short_name"`
-	Types     []string
+	Name      string   `json:"long_name"`
+	ShortName string   `json:"short_name"`
+	Types     []string `json:"types"`
 }
 
 type Geometry struct {
-	Bounds   Bounds
-	Location Point
-	Type     string
-	Viewport Bounds
+	Bounds   *Bounds `json:"bounds,omitempty"`
+	Location Point  `json:"location"`
+	Type     string `json:"location_type"`
+	Viewport Bounds `json:"viewport"`
```

Comment:
Viewport is mandatory, but bounds is optionally returned by google api: https://developers.google.com/maps/documentation/javascript/geocoding

With Bounds not being a pointer, omitempty option wasn't omitting the result and send the default empty Bound, which probably was the reason for the origin being 0,0 on live.